portal-api/src/v1/controllers/deal-status/update-facility-cover-start-dates.js

```diff
@@ -1,42 +1,56 @@
 const CONSTANTS = require('../../../constants');
+const { getNowAsEpoch } = require('../../helpers/date');
 const facilitiesController = require('../facilities.controller');
 
+/**
+ * Updates the cover start dates of facilities in a deal.
+ * @param {object} user - The user object representing the user performing the update.
+ * @param {object} deal - The deal object containing the facilities to be updated.
+ * @returns {object} - The modified deal object with updated facility cover start dates.
+ */
 const updateFacilityCoverStartDates = async (user, deal) => {
-  const modifiedDeal = deal;
+  try {
+    const modifiedDeal = { ...deal };
 
-  if (modifiedDeal.facilities.length) {
-    modifiedDeal.facilities.forEach(async (facilityId) => {
-      const facility = await facilitiesController.findOne(facilityId);
+    if (!modifiedDeal.facilities || !modifiedDeal.facilities?.length) {
+      console.error('No facilities found in deal %s', modifiedDeal._id);
+      return modifiedDeal;
```

Comment:
I don't know where this is called, but should this ever happen in the system or should we throw an error here that can be caught?

Reply:
Good shout, have replaced with `throw new Error(...)` and have also subseuqently updated the unit test cases.